src/protocols/rdp/rdp_settings.c

```diff
@@ -775,6 +783,14 @@ guac_rdp_settings* guac_rdp_parse_args(guac_user* user,
     settings->sftp_directory =
         guac_user_parse_args_string(user, GUAC_RDP_CLIENT_ARGS, argv,
                 IDX_SFTP_DIRECTORY, NULL);
+
+    /* Default keepalive value */
+    settings->sftp_keepalive =
+        guac_user_parse_args_int(user, GUAC_RDP_CLIENT_ARGS, argv,
+                IDX_SFTP_KEEPALIVE, 0);
+    if (settings->sftp_keepalive == 1)
+        guac_user_log(user, GUAC_LOG_WARNING, "The minimum allowed "
+                "value for keepalives by libssh2 is 2 seconds.");
```

Comment:
It's important that the reasoning for behavior be documented internally, but citing such internals in the logs will only lead to confusion. To the administrator, they're looking through the logs to debug Guacamole, not its dependencies, and the fact that it's libssh2 that imposes this restriction has no practical implications. All that matters is that the specified keep-alive interval has been automatically rounded up to the minimum value of 2.